addons/account/models/account_tax_report.py

```diff
@@ -227,12 +226,23 @@ def unlink(self):
     def _remove_tags_used_only_by_self(self):
         """ Deletes and removes from taxes and move lines all the
         tags from the provided tax report lines that are not linked
-        to any other tax report lines.
+        to any other tax report lines nor moves.
+        If there is still a link with moves, it will be archived instead
+        to avoid loosing data.
         """
         all_tags = self.mapped('tag_ids')
         tags_to_unlink = all_tags.filtered(lambda x: not (x.tax_report_line_ids - self))
         self.write({'tag_ids': [(3, tag.id, 0) for tag in tags_to_unlink]})
-        self._delete_tags_from_taxes(tags_to_unlink.ids)
+
+        move_with_archive_tag = self.env['account.move.line'].sudo().search([('tax_tag_ids', 'in', tags_to_unlink.ids)], limit=1)
+        # if the tag is still referenced in move lines we archive the tag, else we delete it.
+        if move_with_archive_tag:
+            # unlink archived tags from taxes to avoid old tags to be used anymore.
+            taxes_with_archive_tag = self.env['account.tax.repartition.line'].sudo().search([('tag_ids', 'in', tags_to_unlink.ids)], limit=1)
```

Comment:
```suggestion
            repartition_lines_with_archived_tags = self.env['account.tax.repartition.line'].sudo().search([('tag_ids', 'in', tags_to_unlink.ids)], limit=1)
```
This recordset consists of repartition lines, not taxes (account.tax). Also, archive => archived (more natural and overall clearer)

If you find the name too long, you can say "rep_lines" instead of "repartition_lines"